src/backend/core.nim

```diff
@@ -13,6 +13,33 @@ proc callRPC*(inputJSON: string): string =
 proc callPrivateRPCRaw*(inputJSON: string): string {.raises: [].} =
     result = $status_go.callPrivateRPC(inputJSON)
 
+proc callPrivateRPCWithChainId*(methodName: string, chainId: int, payload = %* []): RpcResponse[JsonNode]
```

Comment:
Would there be a way to merge the function payload of `callPrivateRPCWithChainId` and `callPrivateRPC`. Like move the payload to a private funciton called `doCallPrivateRPC` and pass it the `inputJSON`, so all the checks and logging can be done in only one function?